consensus/mint-client/src/config.rs

```diff
@@ -176,12 +178,21 @@ pub struct MintTxPrefixParams {
 impl MintTxPrefixParams {
     pub fn try_into_mint_tx_prefix(
         self,
+        fog_bits: Option<FogBits>,
         fallback_tombstone_block: impl Fn() -> u64,
     ) -> Result<MintTxPrefix, String> {
-        if let Some(fog_url) = self.recipient.fog_report_url() {
-            return Err(format!("This recipient has a fog url, but minting to fog users is not supported right now: '{}'", fog_url));
-        }
-        let tombstone_block = self.tombstone.unwrap_or_else(fallback_tombstone_block);
+        let mut tombstone_block = self.tombstone.unwrap_or_else(fallback_tombstone_block);
+        let e_fog_hint = if let Some(fog_url) = self.recipient.fog_report_url() {
+            if let Some(fog_bits) = fog_bits {
+                let (e_fog_hint, pubkey_expiry) = fog_bits.get_e_fog_hint(&self.recipient)?;
+                tombstone_block = core::cmp::min(tombstone_block, pubkey_expiry);
+                Some(e_fog_hint)
+            } else {
+                return Err(format!("This recipient has a fog url, but a CSS to validate fog public keys was not supplied: '{}'", fog_url));
+            }
+        } else {
+            None
+        };
```

Comment:
Optional/Equivalent: Use `map` and `ok_or_else`
```suggestion
        let e_fog_hint = self.recipient.fog_report_url().map(|fog_url| {
            let fog_bits = fog_bits.ok_or_else(format!(
                "This recipient has a fog url, but a CSS to validate fog public keys was not supplied: '{}'",
                fog_url,
            ))?;
            let (e_fog_hint, pubkey_expiry) = fog_bits.get_e_fog_hint(&self.recipient)?;
            tombstone_block = tombstone_block.min(pubkey_expiry);
            Ok(e_fog_hint)
        })?;
```

Reply:
I think this will need a call to `tranpose` before the last `?` because the RHS (before ?) has type `Option<Result<...>` rather than `Result<Option<...>>`? looks fine otherwise though, will do